### PR TITLE
Fixes in JWST login

### PR DESCRIPTION
**Pull request by jespinosaar**
Dear Astroquery team,

We realized the login feature in ESA TAP module was not working, as we are now using kwargs. This pull request aims to fix this issue. Thanks in advance for your support!

cc @esdc-esac-esa-int

### CODE REVIEW

**Review by bsipocz — COMMENTED** [review]
While it's very unlikely that we would  routinely find any issues with the login machinery using tests (as we don't have any login credentials in CI, etc.), but I would generally in favour of having some test coverage that directly uses the remote service, rather than fully rely on mocking the expected responses.

This, and the rest of the in-line comments are just comments.

File: docs/esa/jwst/jwst.rst [hunk 1]
```diff
@@ -377,8 +377,8 @@ Once a table is loaded, columns can be inspected
 
   >>> from astroquery.esa.jwst import Jwst
   >>> table = Jwst.load_table('jwst.main')
-  >>> print(*(column.name for column in table.columns), sep="\n")
-  public
+  >>> print(*(column.name for column in table.columns), sep="\n") # doctest: +IGNORE_OUTPUT
+  "public"
```
[the remark this answers is not in the document]
Comment (jespinosaar) [reply docs/esa/jwst/jwst.rst:381]: I skipped these tests, as they will fail when we release the new version. I will create an additional PR when we released, to make them available again.

File: astroquery/esa/jwst/core.py [hunk 2]
```diff
@@ -898,8 +898,7 @@ def get_product(self, *, artifact_id=None, file_name=None):
                 output_file_name = self._query_get_product(artifact_id=artifact_id)
                 err_msg = str(artifact_id)
             except Exception as exx:
-                raise ValueError('Cannot retrieve product for artifact_id '
-                                 + artifact_id + ': %s' % str(exx))
+                raise ValueError('Cannot retrieve product for artifact_id ' + artifact_id + ': %s' % str(exx))
```
[the remark this answers is not in the document]
Comment (jespinosaar) [reply astroquery/esa/jwst/core.py]: Yes, I have applied the same to other strings.

File: astroquery/esa/jwst/core.py [hunk 3]
```diff
@@ -1086,7 +1082,7 @@ def __set_additional_parameters(self, param_dict, cal_level,
 
     def __get_quantity_input(self, value, msg):
         if value is None:
-            raise ValueError("Missing required argument: '"+str(msg)+"'")
+            raise ValueError(f"Missing required argument: '{str(msg)}'")
```
[the remark this answers is not in the document]
Comment (jespinosaar) [reply astroquery/esa/jwst/core.py]: Thanks for catching this

**Comment by jespinosaar** [comment]
I agree that it would be a good idea to include remote tests for the login, but we need to find a secure way of including this test user. Anyway, I think we can include it in a different PR.

Everything is passing now.

**Comment by jespinosaar** [comment]
Ok, finally all the checks have passed!

**Comment by jespinosaar** [comment]
Hi all, any news on this? Is it necessary to include more improvements? If not, can we merge it?

On the other hand, I think the pre-release procedure is not working as expected, can we create a new version after this is merged?

Thanks in advance!

**Review by keflavich — APPROVED** [review]
@bsipocz lgtm.  There's one minor unaddressed comment in the docs, but I think since tests are passing, it is an acceptable change.   Mostly this is fixing a real issue, iiuc, and adding tests, with many scattered style changes mixed in.

**Comment by bsipocz** [comment]
Oh, I'm sorry this has fallen through the cracks. Last time I checked something didn't really work with the remote-data part, let me get back to that today, maybe push a commit and get this merged.

**Comment by jespinosaar** [comment]
> Oh, I'm sorry this has fallen through the cracks. Last time I checked something didn't really work with the remote-data part, let me get back to that today, maybe push a commit and get this merged.

Ok! just executed the remote tests and it seems to be working... Anyway, please let me know if you need anything else.

Related to the pre-released version, is this working? Could you please create a new version when merging this PR? thanks in advance!

File: astroquery/esa/jwst/tests/test_jwstdata.py [hunk 4]
```diff
@@ -57,3 +58,13 @@ def test_get_product(self):
...
+
+
+@pytest.mark.remote_data
+class RemoteData:
+
+    def test_login_error(self):
+        jwst = JwstClass()
+        with pytest.raises(HTTPError) as err:
+            jwst.login(user="dummy", password="dummy")
+        assert "Unauthorized" in err.value.args[0]
```
Comment (bsipocz) [inline astroquery/esa/jwst/tests/test_jwstdata.py:70]: This class is not being picked up for testing.

(I'll push a fix for it)

**Review by bsipocz — APPROVED** [review]
One minor fix, and this is good to go. Thanks @jespinosaar for the patience.

**Comment by bsipocz** [comment]
I also needed to rebase this, so fixes to the docs builds are being picked up.

**Comment by jespinosaar** [comment]
Many thank for your support @bsipocz, as always! and for the pre-release!

**Pull request merged**